src/App.vue

```diff
@@ -1,5 +1,11 @@
 <script setup lang="ts">
 import TheHeader from "@/components/TheHeader.vue";
+import { useMovieStore } from "@/stores/movies";
+import { onMounted } from "vue";
+const movieStore = useMovieStore();
+onMounted(async () => {
```

Comment:
In general I would suggest keeping only global app-wide code in App component. If something is not used everywhere, it's usually better to keep it where it's used. If those movies are only used in movies page, you can move fetching them to `MoviesPage`

Reply:
Idea was to have them prefetched but more then happy to move it to movies page - it makes logical sense

Reply:
That's a valid idea and it's absolutely ok to implement it. As long as you're aware of the trade-off, that this will also cause loading the data every time we enter the App even when we don't visit movies page.

To help making your decision you can also consider whether this data is going to be used on other pages, maybe there is going to be a section with new shows on the front page. Or whether this data is loaded once and stays the same or whether it changes and should be up to date when user visits movies page.

There is no wrong choice here and final decision is up to you :)